packages/slider/src/Slider.ts

```diff
@@ -243,203 +253,60 @@ export class Slider extends Focusable {
         `;
     }
 
-    private renderHandle(): TemplateResult {
+    private renderTrackSegment(start: number, end: number): TemplateResult {
+        if (this.variant === 'ramp') {
+            return html``;
+        }
         return html`
             <div
-                id="handle"
-                style=${this.handleStyle}
-                @manage=${streamingListener(
-                    { type: 'pointerdown', fn: this.handlePointerdown },
-                    { type: 'pointermove', fn: this.handlePointermove },
-                    {
-                        type: ['pointerup', 'pointercancel'],
-                        fn: this.handlePointerup,
-                    }
-                )}
+                class="track"
+                style=${styleMap(this.trackSegmentStyles(start, end))}
                 role="presentation"
-            >
-                <input
-                    type="range"
-                    id="input"
-                    value=${this.value}
-                    step=${this.step}
-                    min=${this.min}
-                    max=${this.max}
-                    aria-disabled=${ifDefined(
-                        this.disabled ? 'true' : undefined
-                    )}
-                    aria-valuetext=${this.ariaValueText}
-                    @change=${this.onInputChange}
-                    @focus=${this.onInputFocus}
-                    @blur=${this.onInputBlur}
-                />
-            </div>
+            ></div>
         `;
     }
 
     private renderTrack(): TemplateResult {
+        const segments = this.handleController.trackSegments();
         return html`
             <div @pointerdown=${this.handleTrackPointerdown}>
                 <div id="controls">
-                    ${this.renderTrackLeft()} ${this.renderRamp()}
-                    ${this.renderTicks()} ${this.renderHandle()}
-                    ${this.renderTrackRight()}
+                    ${this.renderTrackSegment(...segments[0])}
+                    ${this.renderRamp()} ${this.renderTicks()}
+                    ${this.handleController.render()}
+                    ${segments
+                        .slice(1)
```

Comment:
What are we slicing off here? Is there a way to do that in the controller?

Reply:
So, the CSS assumes that the first and last things rendered are track segments. So, I render the first segment explicitly and then render the remaining ones at the end. The `slice(1)` just takes all of the remaining segments after the first one.

Reply:
Interesting. I'd want to check whether slicing vs skipping had any effect on the way the template was addressed in subsequent renders, being the slice (IIRC) makes a new array identity every time.

Reply:
I reworked this to make sure that the various parts of rendered HTML have appropriate ids. I am not sure I gained much in performance. My ad hoc testing showed there was possibly a small difference.